sdk/cosmos/azure_data_cosmos/src/pipeline/mod.rs

```diff
@@ -191,3 +194,39 @@ pub(crate) fn create_base_query_request(url: Url, query: &Query) -> azure_core::
     request.set_json(query)?;
     Ok(request)
 }
+
+trait ResponseExt {
+    async fn success(self) -> azure_core::Result<Self>
+    where
+        Self: Sized;
+}
+
+impl<T, F> ResponseExt for Response<T, F> {
+    async fn success(self) -> azure_core::Result<Self> {
+        if self.status().is_success() {
+            Ok(self)
+        } else {
+            RawResponse::from(self).success().await.map(Into::into)
+        }
+    }
+}
+
+impl ResponseExt for RawResponse {
```

Comment:
@chlowell this is a good idea we might consider in `azure_core`. Combined with #2725 we could also use something like this to wrap standard Azure errors. We'd have to implement it (foreign trait otherwise), but could also allow clients to pass in custom success codes or something, much like .NET does for their DPG clients and HTTP status code categorizations.